Sources/Hyperspace/Service/Network/NetworkServiceProtocol.swift

```diff
@@ -71,3 +71,33 @@ public protocol NetworkServiceProtocol {
     /// Cancels all currently running tasks
     func cancelAllTasks()
 }
+
+// MARK: - Equatable Implementations
+
+extension NetworkServiceError: Equatable {
+    
+    public static func == (lhs: NetworkServiceError, rhs: NetworkServiceError) -> Bool {
+        switch (lhs, rhs) {
+        case (.unknownError(let lhsError), .unknownError(let rhsError)):
+            return lhsError?.localizedDescription == rhsError?.localizedDescription
```

Comment:
This is the only piece of this that makes me a little uneasy. Unfortunately, I don't know of a better way to check for equality of two objects conforming to `Error`, but I think this should be safe for 99% of use cases. We were already doing this in our test code for `MockBackendServiceError`.

Reply:
We could also just treat all `.unknownError` as equal, regardless of associated `Error`? Maybe?

Reply:
We could, but I saw some recommendations against that on the [Swift forums](https://forums.swift.org/t/making-error-sub-enums-equatable/5878/6).